TestCCLLogic_redact_descriptor failed with extra createAsOfTime field

pkg/ccl/logictestccl/tests/fakedist-vec-off/fakedist-vec-off_test.TestCCLLogic_redact_descriptor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8656202?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8656202?buildTab=artifacts#/) on master @ [e51ffa013c81212870891001f0328912550fa75d](https://github.com/cockroachdb/cockroach/commits/e51ffa013c81212870891001f0328912550fa75d):


```
                        "partitioning": {
                            "numColumns": 1,
                            "range": [
                                {
                                    "fromInclusive": "Xw==",
                                    "name": "p1",
                                    "toExclusive": "Xw=="
                                }
                            ]
                        },
                        "sharded": {},
                        "storeColumnIds": [
                            2
                        ],
                        "storeColumnNames": [
                            "b"
                        ],
                        "unique": true,
                        "version": 4
                    },
                    "privileges": {
                        "ownerProto": "root",
                        "users": [
                            {
                                "privileges": "2",
                                "userProto": "admin",
                                "withGrantOption": "2"
                            },
                            {
                                "privileges": "2",
                                "userProto": "root",
                                "withGrantOption": "2"
                            }
                        ],
                        "version": 2
                    },
                    "replacementOf": {
                        "time": {}
                    },
                    "unexposedParentSchemaId": 105,
                    "version": "9"
                }
            }
[06:29:13] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3694/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/redact_descriptor with config fakedist-vec-off: 4 tests, 1 failures
    logic.go:4012: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3694/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/redact_descriptor:172: error while processing
    logic.go:4012: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3694/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/redact_descriptor:172: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/da6a30601c9144091e05fdc42b4683e0/logTestCCLLogic_redact_descriptor2641124468
--- FAIL: TestCCLLogic_redact_descriptor (1.96s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_redact_descriptor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24411

--- a/pkg/ccl/logictestccl/testdata/logic_test/redact_descriptor
+++ b/pkg/ccl/logictestccl/testdata/logic_test/redact_descriptor
@@ -16,20 +16,24 @@ CREATE VIEW redacted_descriptors AS
         jsonb_pretty(
             regexp_replace(
                 regexp_replace(
-                    (
-                        crdb_internal.pb_to_json(
-                            'desc',
-                            crdb_internal.redact_descriptor(
-                                descriptor
+                    regexp_replace(
+                        (
+                            crdb_internal.pb_to_json(
+                                'desc',
+                                crdb_internal.redact_descriptor(
+                                    descriptor
+                                )
                             )
-                        )
-                    )::STRING,
-                    '"createdAtNanos": "[0-9]+"',
-                    '"createdAtNanos": "0"',
-                    'g'
+                        )::STRING,
+                        '"createdAtNanos": "[0-9]+"',
+                        '"createdAtNanos": "0"',
+                        'g'
+                    ),
+                    '"wallTime": "[0-9]+"',
+                    '"wallTime": "0"'
                 ),
-                '"wallTime": "[0-9]+"',
-                '"wallTime": "0"'
+               '"logical": [0-9]+[,]*',
+               ''
             )::JSONB
         )
             AS descriptor

--- a/pkg/sql/logictest/testdata/logic_test/redact_descriptor
+++ b/pkg/sql/logictest/testdata/logic_test/redact_descriptor
@@ -5,20 +5,24 @@ CREATE VIEW redacted_descriptors AS
         jsonb_pretty(
             regexp_replace(
                 regexp_replace(
-                    (
-                        crdb_internal.pb_to_json(
-                            'desc',
-                            crdb_internal.redact_descriptor(
-                                descriptor
+                    regexp_replace(
+                        (
+                            crdb_internal.pb_to_json(
+                                'desc',
+                                crdb_internal.redact_descriptor(
+                                    descriptor
+                                )
                             )
-                        )
-                    )::STRING,
-                    '"createdAtNanos": "[0-9]+"',
-                    '"createdAtNanos": "0"',
-                    'g'
+                        )::STRING,
+                        '"createdAtNanos": "[0-9]+"',
+                        '"createdAtNanos": "0"',
+                        'g'
+                    ),
+                    '"wallTime": "[0-9]+"',
+                    '"wallTime": "0"'
                 ),
-                '"wallTime": "[0-9]+"',
-                '"wallTime": "0"'
+               '"logical": [0-9]+[,]*',
+               ''
             )::JSONB
         )
             AS descriptor

